ts/features/bonusVacanze/screens/ActiveBonusScreen.tsx

```diff
@@ -78,19 +84,37 @@ const ActiveBonusScreen: React.FunctionComponent<Props> = (props: Props) => {
     });
   };
 
+  const shareQR = async (content: string, code: string) => {
```

Comment:
A functional approach to Promise that could be fail (then+catch) is using [TaskEither](https://gcanti.github.io/fp-ts/modules/TaskEither.ts.html)
It's the same but less code and more functional

```typescript
import { tryCatch } from "fp-ts/lib/TaskEither";

const shareContent = (content: string, message?: string) => {
    return tryCatch(
      () =>
        Share.open({
          url: `data:image/png;base64,${content}`,
          message
        }),
      errorMsg => new Error(String(errorMsg))
    );
  };

  const shareQR = async (content: string, code: string) => {
    const shared = await shareContent(content, code).run();
    if (shared.isLeft()) {
      showToast(I18n.t("global.genericError"));
    }
  };
```

If you find this as a better implementation could be **shareContent** an util function inside `utils/share.fs`?

Reply:
yes it could be useful, do you think it could be better in `bonusVacanze/utils` or in the main folder?

Reply:
I guess it could be use for the whole application -> `so ts/utils/share.ts`